rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -354,6 +354,29 @@ rmw_publisher_get_actual_qos(
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+rmw_publisher_assert_liveliness(const rmw_publisher_t * publisher)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
+
+  auto info = static_cast<OpenSpliceStaticPublisherInfo *>(publisher->data);
+  if (nullptr == info) {
+    RMW_SET_ERROR_MSG("publisher internal data is invalid");
+    return RMW_RET_ERROR;
+  }
+  if (nullptr == info->topic_writer) {
+    RMW_SET_ERROR_MSG("publisher internal datawriter is invalid");
+    return RMW_RET_ERROR;
+  }
+
+  if (info->topic_writer->assert_liveliness() != DDS::RETCODE_OK) {
```

Comment:
Same comment as @nburek, should probably use the DDS to RMW return code function.

Reply:
Done.